### ISSUE TITLE
Implement Level Select 

### ISSUE THREAD

**Issue by david-pfx** [ClementSparrow/Pattern-Script#61]
1. Upward compatible with PS+
2. Modular, basis for future alternative versions

**Comment by ClementSparrow** [comment]
3. Use the level-naming features described in https://clementsparrow.github.io/Pattern-Script/src/Documentation/levels.html

**Comment by david-pfx** [comment]
1. On closer inspection, the PS+ model is heavily dependent on SECTION, which brings in a whole new can of worms. Your rewrite of LevelState seems a quite different approach, not easily reconciled.
2. Modular won't work as long as files are loaded from the HTML. Just adding a file is way too much work and too intrusive. It needs to be refactored to use ES5 require or similar, so files are loaded in the JS. [That will fix some other issues too, where an error in one file triggers an unrelated error in an unrelated file, due to the order of file loading.]
3. Level names are fine and the menus are working, but I'm having problems with the 'box' stuff. There doesn't seem to be an existing API to load a random level.

**Comment by david-pfx** [comment]
I've got it all working. Do you have a program with an insane combination of levels (at least 30), level names and messages I can try it out on?

**Comment by ClementSparrow** [comment]
No, unfortunately no game seems to have been done yet with that many levels and level names and messages.
